ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -1166,24 +1191,57 @@ run tracers tracersExtra args argsExtra apps appsExtra = do
+                   -> StrictTVar IO (PeerSelectionState RemoteAddress ntnconn)
+                   -> PeerMetrics IO RemoteAddress
                    -> IO ()
 #ifdef POSIX
-                 diInstallSigUSR1Handler = \connectionManager -> do
+                 diInstallSigUSR1Handler = \connectionManager dbgStateVar metrics -> do
                    _ <- Signals.installHandler
                      Signals.sigUSR1
                      (Signals.Catch
                        (do state <- atomically $ readState connectionManager
                            traceWith (dtConnectionManagerTracer tracersExtra)
                                      (TrState state)
+                           PeerSelectionState{..} <- atomically $ readTVar dbgStateVar
+                           now <- getMonotonicTime
+                           up <- atomically $ upstreamyness metrics
+                           bp <- atomically $ fetchynessBlocks metrics
+                           let dbgState = DebugPeerSelectionState {
+                                          dpssTargets = targets
+                                        , dpssLocalRootPeers = localRootPeers
+                                        , dpssPublicRootPeers = publicRootPeers
+                                        , dpssBigLedgerPeers = bigLedgerPeers
+                                        , dpssKnownPeers = knownPeers
+                                        , dpssEstablishedPeers =
+                                            EstablishedPeers.toSet establishedPeers
+                                        , dpssActivePeers = activePeers
+                                        , dpssPublicRootBackoffs = publicRootBackoffs
+                                        , dpssPublicRootRetryTime = publicRootRetryTime
+                                        , dpssInProgressPublicRootsReq = inProgressPublicRootsReq
+                                        , dpssBigLedgerPeerBackoffs = bigLedgerPeerBackoffs
+                                        , dpssBigLedgerPeerRetryTime = bigLedgerPeerRetryTime
+                                        , dpssInProgressBigLedgerPeersReq =
+                                            inProgressBigLedgerPeersReq
+                                        , dpssInProgressPeerShareReqs = inProgressPeerShareReqs
+                                        , dpssInProgressPromoteCold = inProgressPromoteCold
+                                        , dpssInProgressPromoteWarm = inProgressPromoteWarm
+                                        , dpssInProgressDemoteWarm = inProgressDemoteWarm
+                                        , dpssInProgressDemoteHot = inProgressDemoteHot
+                                        , dpssInProgressDemoteToCold = inProgressDemoteToCold
+                                        , dpssUpstreamyness = up
+                                        , dpssFetchynessBlocks = bp
+                                        }
```

Comment:
Could you abstract to a pure function:

```
makeDebugPeerSelectionState :: PeerSelectionState -> ... -> DebugPeerSelectionState
...
```

This function should be defined next to `DebugPeerSelectionState` type.